### PR TITLE
CMake options for enabling Clang sanitizer

### PR DESCRIPTION
**Pull request by bilke**
- `OGS_ADDRESS_SANITIZER` for [AddressSanitizer](http://clang.llvm.org/docs/AddressSanitizer.html)
- `OGS_UNDEFINED_BEHAVIOR_SANITIZER` for [UndefinedBehaviorSanitizer](http://clang.llvm.org/docs/UndefinedBehaviorSanitizer.html)

Detects #735.

I will setup Jenkins jobs after merge.

### CODE REVIEW

File: scripts/cmake/ClangSanitizer.cmake [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+option(OGS_ADDRESS_SANITIZER OFF "Use Clangs AddressSanitizer")
+option(OGS_UNDEFINED_BEHAVIOR_SANITIZER OFF "Use Clangs UndefinedBehaviorSanitizer")
+
+if(OGS_ADDRESS_SANITIZER)
+	set(SANITIZE_FLAG_VALUE "address")
+	set(ADDITIONAL_FLAGS "-fno-omit-frame-pointer")
+endif()
+
+if(OGS_UNDEFINED_BEHAVIOR_SANITIZER)
+	set(SANITIZE_FLAG_VALUE "${SANITIZE_FLAG_VALUE},undefined")
```
Comment (endJunction) [inline scripts/cmake/ClangSanitizer.cmake]: I'm not sure if both sanitizers can be use at the same time. Did not find anything on the [clang docu](http://clang.llvm.org/docs/UndefinedBehaviorSanitizer.html) but usually they specify if sanitizers can be combined (like the [LeakSanitizer](http://clang.llvm.org/docs/LeakSanitizer.html)).

Comment (endJunction) [inline scripts/cmake/ClangSanitizer.cmake]: Please add `-fsanitize=integer` too for full strength.

Comment (bilke) [inline scripts/cmake/ClangSanitizer.cmake]: [Docs](http://clang.llvm.org/docs/UsersManual.html#id33) state that:

> It is not possible to combine more than one of the -fsanitize=address, -fsanitize=thread, and -fsanitize=memory checkers in the same program.

So combining `undefined` with `address` should be ok.

Comment (bilke) [inline scripts/cmake/ClangSanitizer.cmake]: Should `integer` be an independent CMake option or should it be added with `undefined`? And would it then be `-fsanitize=undefined,integer`?

Comment (endJunction) [inline scripts/cmake/ClangSanitizer.cmake]: Shall be added to `undefined`. Looks like they can be just comma-separated.

**Pull request merged**